crypto/ec/ecx_meth.c

```diff
@@ -406,16 +406,18 @@ static int ecx_generic_import_from(const OSSL_PARAM params[], void *vpctx,
 
 static int ecx_pkey_copy(EVP_PKEY *to, EVP_PKEY *from)
 {
-    ECX_KEY *ecx = from->pkey.ecx;
+    ECX_KEY *ecx = from->pkey.ecx, *dupkey = NULL;
     int ret;
 
-    /* We can do just up-ref as ECX keys are immutable */
-    if (ecx != NULL && !ossl_ecx_key_up_ref(ecx))
-        return 0;
+    if (ecx != NULL) {
+        dupkey = ossl_ecx_key_dup(ecx);
```

Comment:
Is this right to do a dup here ?

Reply:
Yes, why not? The up_ref was a hack. And yes, the up_ref would work because there is no public interface for legacy ECX keys that would modify their value. But semantically the dup should be there and as we now have it why not use it.